picard/tagger.py

```diff
@@ -360,9 +445,106 @@ def load_to_picard(self, items):
 
-        if parsed_items:
+        for command in parsed_items.commands:
+            self.handle_command(command)
+
+        if parsed_items.non_executable_items():
             self.bring_tagger_front()
 
+    def get_album_pane_tracks(self):
+        for album in self.albums.values():
+            for track in album.iterfiles():
+                yield track
+
+    def get_all_file_objects(self):
+        yield from self.unclustered_files.files
+        yield from self.get_album_pane_tracks()
+        yield from self.clusters.iterfiles()
+
+    def _init_remote_commands(self):
+        self.commands = {name: getattr(self, remcmd.method_name) for name, remcmd in REMOTE_COMMANDS.items()}
+
+    def handle_command(self, command):
+        cmd, *args = command.split(' ', 1)
+        argstring = next(iter(args), "")
+        cmd = cmd.upper()
+        log.debug("Executing command: %r", cmd)
+        try:
+            thread.to_main(self.commands[cmd], argstring.strip())
+        except KeyError:
+            log.error("Unknown command: %r", cmd)
+
+    def handle_command_cluster(self, argstring):
+        self.cluster(self.unclustered_files.files)
+
+    def handle_command_fingerprint(self, argstring):
+        for album_name in self.albums:
+            self.analyze(self.albums[album_name].iterfiles())
+
+    def handle_command_lookup(self, argstring):
+        self.autotag(self.unclustered_files.files)
+
+    def handle_command_lookup_cd(self, argstring):
+        disc = Disc()
+        devices = get_cdrom_drives()
+
+        if not argstring:
+            if devices:
+                device = devices[0]
+            else:
+                device = None
+        elif argstring in devices:
+            device = argstring
+        else:
+            thread.run_task(
+                partial(self._parse_disc_ripping_log, disc, argstring),
+                partial(self._lookup_disc, disc),
+                traceback=self._debug)
+            return
+
+        thread.run_task(
+            partial(disc.read, encode_filename(device)),
+            partial(self._lookup_disc, disc),
+            traceback=self._debug)
+
+    def handle_command_quit(self, argstring):
+        self.exit()
+        self.quit()
+
+    def handle_command_remove(self, argstring):
+        for file in self.get_all_file_objects():
+            if argstring == file.filename:
+                file.remove()
+                return
+
+    def handle_command_remove_all(self, argstring):
+        for file in self.get_all_file_objects():
+            file.remove()
+
+    def handle_command_remove_saved(self, argstring):
+        for track in self.get_album_pane_tracks():
+            if track.state == File.NORMAL:
+                track.remove()
+
+    def handle_command_save_matched(self, argstring):
+        for album in self.albums.values():
+            for track in album.iter_correctly_matched_tracks():
+                track.files[0].save()
```

Comment:
Just to clarify, `album.iter_correctly_matched_tracks()` returns tracks with only one file attached, which is why saving only `track.files[0]` is appropriate, right?